frontend/src/features/comparisons/Comparison.tsx

```diff
@@ -143,12 +146,31 @@ const Comparison = ({
       let autoUidA = null;
       let autoUidB = null;
 
-      if (!uidA && autoFillSelectorA) {
-        autoUidA = await nextSuggestion(uidA, uidB, pollName);
+      // At this moment, the entities still present in selectorAHistory and
+      // selectorBHistory come from a previous comparison session. During this
+      // last session they may have been compared enough times to not be
+      // considered as suggestion by the API anymore. To avoid suggesting
+      // entities that don't match the suggestion strategies used by the API,
+      // we clear the history of all selectors.
+      selectorAHistory.clear();
+      selectorBHistory.clear();
```

Comment:
Does that mean the history is always reset when the comparison page is reloaded?
If that's the case, it could be more simple and explicit to initialize the history as `useRef()` in this component?

Reply:
> Does that mean the history is always reset when the comparison page is reloaded?

Exactly.

> If that's the case, it could be more simple and explicit to initialize the history as useRef() in this component?

Do you mean initializing it inside the component `Comparison` instead of importing it as a global variable?

Reply:
Addressed by https://github.com/tournesol-app/tournesol/pull/1954/commits/3117722a8e20280c1912b5bb32280886f7125540